spark/src/main/scala/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -274,35 +275,64 @@ class DeltaLog private(
   /**
    * Get all actions starting from "startVersion" (inclusive). If `startVersion` doesn't exist,
    * return an empty Iterator.
+   * Callers are encouraged to use the other override which takes the endVersion if available to
+   * avoid I/O and improve performance of this method.
```

Comment:
Do we have (m)any internal Delta call sites of these "deprecated" method signatures? 
Or have they all been updated and we're only worried about external callers?

Reply:
We do have more call-sites where end is not available. e.g. `DeltaSourceCDCSupport` uses this API.

Reply:
And those are cases where we don't want to filter by end, right? Things like "get everything available" ?